lib/base/tlsstream.cpp

```diff
@@ -33,6 +33,11 @@ std::shared_ptr<X509> UnbufferedAsioTlsStream::GetPeerCertificate()
 	return std::shared_ptr<X509>(SSL_get_peer_certificate(native_handle()), X509_free);
 }
 
+STACK_OF(X509) *UnbufferedAsioTlsStream::GetPeerCertificateChain()
+{
+	return SSL_get_peer_cert_chain(native_handle());
+}
+
```

Comment:
... hopefully(!) supplies a such chain if(!) it's included.

Reply:
This one has no shared pointer, but IMAO that's OK, see NOTES in https://www.openssl.org/docs/man1.1.1/man3/SSL_get_peer_cert_chain.html

Reply:
Yes, that function just returns a bundle of all the certificates the icinga-agent has put on the wire. Theoretically it could be `NULL` (which would be ok), but in my tests it will always just the whole stack of all certificates the agent knows from its `ca.crt` and `cert.pem`.

As far as I know there is also no need to free this pointer, since it is internally not referenced by OpenSSL and gets invalided once the connection drops.